### PR TITLE
Adds tests against the wycheproof test vectors

### PR DESCRIPTION
**Pull request by andozw**
Along the same spirit as the secp256k1 PR 1245. This tests against all 463 vectors but includes exclusions for 3 of them. Comments include the details outlining why each was excluded.

The json and c code was generated using the included python code (utils/tests_wycheproof_generate.py).

We decided not to duplicate this generation code for now.

### CODE REVIEW

**Comment by ivmaykov** [comment]
It looks like the subzero core binary failed to start in the background in the SignTx regression test action. Hard to tell what's going on, we probably need to improve the logic in the action so it's easier to get logs output. Did you run it locally with ASAN + UBSAN without any issues?

File: core/src/checks/wycheproof_checks.c
Comment (ivmaykov) [inline core/src/checks/wycheproof_checks.c:176]: nit: don't need this anymore, I fixed it in #626

File: core/trezor-crypto/ecdsa.c
Comment (ivmaykov) [inline core/trezor-crypto/ecdsa.c:27]: remove

**Comment by andozw** [comment]
> It looks like the subzero core binary failed to start in the background in the SignTx regression test action. Hard to tell what's going on, we probably need to improve the logic in the action so it's easier to get logs output. Did you run it locally with ASAN + UBSAN without any issues?

Ran locally with `time(1)`. Took about 0.5 seconds without ASAN/UBSAN flags and about 2.6 seconds with ASAN/UBSAN. 

Added sleeps in the CI script and the tests execute successfully which confirms the theory.

File: core/src/checks/wycheproof_checks.c
Comment (andozw) [reply core/src/checks/wycheproof_checks.c:126]: This is what I did in the bitcoin core version as well
https://github.com/bitcoin-core/secp256k1/blob/e5de45460953c8ae16521b1928ac14de218998a3/src/tests.c#LL7314C1-L7314C1
I wanted to have the smallest scope possible for the variables defined in the include file

File: core/src/checks/wycheproof_checks.c
Comment (andozw) [reply core/src/checks/wycheproof_checks.c:129]: c is great `%zu`

**Comment by andozw** [comment]
> * old binary size
> * new binary size
> * difference in kB

Old binary size: 546916 bytes
New binary size: 629876 bytes
Difference ~82.9 kB

File: .github/workflows/signtx-test.yml
Comment (ivmaykov) [inline .github/workflows/signtx-test.yml:66]: how do we know that 10 seconds gives us a good safety margin and won't result in somewhat flaky actions? (or do we know that?)

File: core/src/checks/self_checks.c
Comment (ivmaykov) [inline core/src/checks/self_checks.c:57]: super nit: prefer more consistency between declaration order in checks.h and execution order in self_checks.c

File: core/src/checks/wycheproof_checks.c
Comment (ivmaykov) [inline core/src/checks/wycheproof_checks.c:19]: nit: is there a named constant for this `64` somewhere? If not, should we add one?

File: core/src/checks/wycheproof_checks.c
Comment (ivmaykov) [reply core/src/checks/wycheproof_checks.c]: oh right, "%zu" not "%z". My bad.

**Comment by ivmaykov** [comment]
Approving with some nits, please fix them before merging.

**Review by ivmaykov — APPROVED** [review]
Actually approving this time.

File: .github/workflows/signtx-test.yml
Comment (andozw) [reply .github/workflows/signtx-test.yml:66]: guess and check

File: core/src/checks/wycheproof_checks.c
Comment (andozw) [reply core/src/checks/wycheproof_checks.c:19]: i'll do it as a follow up PR

File: .github/workflows/signtx-test.yml
Comment (andozw) [reply .github/workflows/signtx-test.yml:66]: feels like a goldilocks amount. long enough to allow subzero to boot but not too long to delay CI. We can always change it as needed.

**Review by ivmaykov — APPROVED** [review]
LGTM

**Pull request merged**
